cf/env/env.go

```diff
@@ -29,20 +31,31 @@ func New(delegate server.Environment) server.Environment {
 }
 
 type cfEnvironment struct {
-	cfEnv    *cfenv.App
+	cfEnv *cfenv.App
 	server.Environment
 }
 
-func (e *cfEnvironment) Load() error {
-	var err error
+func (e *cfEnvironment) Load() (err error) {
 	if err = e.Environment.Load(); err != nil {
-		return err
+		return
+	}
+	var postgreServiceName string
+	if serviceName := e.Environment.Get("db.name"); serviceName != nil {
+		postgreServiceName = serviceName.(string)
+	} else {
+		logrus.Warning("No PostgreSQL service name found")
+		return
 	}
 	if e.cfEnv, err = cfenv.Current(); err != nil {
-		return err
+		return
```

Comment:
shouldnt we return the error in case we try to load the cf env and it actually fails?

Reply:
That's what's happening here (we change the value of err which is returned always). I think it will be more clear if it is return err though,

Reply:
oh right, you are using the (err error)  named return syntax - i didnt pay attention to the method signature